state-transition/core/state_processor_staking.go

```diff
@@ -52,6 +52,9 @@ func (sp *StateProcessor[
 			sp.cs.MaxDepositsPerBlock(), len(deposits),
 		)
 	}
+	if err := st.SetEth1Data(blk.GetBody().GetEth1Data()); err != nil {
+		return err
+	}
```

Comment:
`Eth1Data` here is tied to the deposits coming with the block. We should set it in the state only after the deposit validation (and processing), see [here](https://github.com/berachain/beacon-kit/blob/d43b1bd7516a2ff656cff8f74391f03ebb91afef/state-transition/core/state_processor_staking.go#L64-L65)

Reply:
sure